src/Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests/PullDeliveryTests.cs

```diff
@@ -47,12 +49,55 @@ public async Task TestPublishAndReceiveEvent()
         await runTask;
     }
 
+    [Theory]
+    [InlineData(10)]
+    [InlineData(100)]
+    [InlineData(1000)]
+    [InlineData(10000)]
+    public async Task TestHeavyLoadOfEvents(int eventCount)
```

Comment:
I don't think the emulator can return multiple events at the moment. I don't know if this impacts the tests.

Reply:
It probably lowers the overall performance gains, but the test only makes sure that we are running faster than serially (aka at a faster event/second rate than 1000ms / "slow task" delay).

In the end I don't think it could become flaky unless we used a lower "slow task" delay and reading from the emulator becomes the bottleneck. With ~250ms it should be safe.

Still it could be a nice improvement on the emulator to support this!